src/Arcomage/Arcomage.h

```diff
@@ -11,78 +11,102 @@
 
 class GraphicsImage;
 
+enum class ArcomageCompareParam {
+    INVALID = 0,
+    NO_COMPARE = 1,
+    LESSER_QUARRY = 2,
+    LESSER_MAGIC = 3,
+    LESSER_ZOO = 4,
+    EQUAL_QUARRY = 5,
+    EQUAL_MAGIC = 6,
+    EQUAL_ZOO = 7,
+    GREATER_QUARRY = 8,
+    GREATER_MAGIC = 9,
+    GREATER_ZOO = 10,
+    NO_WALL = 11,
+    HAVE_WALL = 12,
+    ENEMY_HAS_NO_WALL = 13,
+    ENEMY_HAS_WALL = 14,
+    LESSER_WALL = 15,
+    LESSER_TOWER = 16,
+    EQUAL_WALL = 17,
+    EQUAL_TOWER = 18,
+    GREATER_WALL = 19,
+    GREATER_TOWER = 20
+};
```

Comment:
Codestyle:

`enum class X` should be followed by `using enum X` & used w/o the `X::` prefix.

I suggest naming the enum as `ArcomageCheck`, then renaming the enum values like this:
```
CHECK_INVALID = 0,
CHECK_NONE = 1,
CHECK_LESSER_QUARRY = 2,
CHECK_LESSER_MAGIC = 3,
CHECK_LESSER_ZOO = 4,
...
```